test/hotspot/jtreg/runtime/cds/appcds/jigsaw/ExactOptionMatch.java

```diff
@@ -0,0 +1,96 @@
+ *          --add-exports, --add-modules, and -enable-native-access.
+ * @bug 8352437
+ * @modules java.logging
+ * @requires vm.cds.write.archived.java.heap
+ * @requires vm.flagless
+ * @library /test/lib /test/hotspot/jtreg/runtime/cds/appcds
+ * @run driver ExactOptionMatch
+ */
+
+import jdk.test.lib.process.OutputAnalyzer;
+
+public class ExactOptionMatch {
+    record Option(String cmdLine, String property, String valueA, String valueB) {}
+
+    static Option[] allOptions = new Option[] {
+        new Option("--add-exports",
+                   "jdk.module.addexports",
+                   "java.base/jdk.internal.misc=ALL-UNNAMED",
+                   "java.base/jdk.internal.misc=ALL-SYSTEM"),
+        new Option("--add-modules",
+                   "jdk.module.addmods",
+                   "java.base",
+                   "java.logging"),
+        new Option("--enable-native-access",
+                   "jdk.module.enable.native.access",
+                   "java.base",
+                   "java.logging"),
+    };
+
+    static final String FMG_DISABLED = "initial full module graph: disabled";
+    static final String FMG_ENABLED = "use_full_module_graph = true";
+
+    public static void main(String[] args) throws Exception {
+        OutputAnalyzer out;
+
+        for (Option o : allOptions) {
+            TestCommon.startNewArchiveName();
+            String archiveName = TestCommon.getCurrentArchiveName();
+
+            // (1) Dump = specified, Run = not specified
+            TestCommon.dumpBaseArchive(archiveName, o.cmdLine(), o.valueA())
+                .shouldHaveExitValue(0);
+
+            TestCommon.execCommon("-Xlog:cds", "--version")
+                .shouldMatch("Mismatched values for property " + o.property() + ": j.*specified during dump time but not during runtime")
+                .shouldContain(FMG_DISABLED);
+
+            // (2) Dump = not specified, Run = specified
+            TestCommon.dumpBaseArchive(archiveName)
+                .shouldHaveExitValue(0);
+
+            TestCommon.execCommon("-Xlog:cds", o.cmdLine(), o.valueA(), "--version")
+                .shouldMatch("Mismatched values for property " + o.property() + ": j.*specified during runtime but not during dump time")
+                .shouldContain(FMG_DISABLED);
+
+            // (3) Dump = specified twice, Run = specified once
+            TestCommon.dumpBaseArchive(archiveName, o.cmdLine(), o.valueA(), o.cmdLine(), o.valueB())
+                .shouldHaveExitValue(0);
+
+            TestCommon.execCommon("-Xlog:cds", o.cmdLine(), o.valueA(), "--version")
+                .shouldMatch("Mismatched values for property " + o.property() + ": runtime.*dump time")
+                .shouldContain(FMG_DISABLED);
+
+            // (4) Dump = specified twice, Run = specified twice (but in different order)
+            //     Should still be able to use FMG (values are sorted by CDS).
```

Comment:
How about add another test case where the values are specified in the same order?

Reply:
Fixed.